### PR TITLE
chore: unify __construct functions

### PR DESCRIPTION
**Pull request by Chartman123**

### CODE REVIEW

**Review by susnux — COMMENTED** [review]
Super nice refactoring!

**Comment by Chartman123** [comment]
I also tried to do it for the DB classes but got lots of errors so I left them as they are for now

File: lib/Controller/ShareApiController.php [hunk 1]
```diff
@@ -55,7 +55,7 @@
 
 class ShareApiController extends OCSController {
 	private IUser $currentUser;
-
+
```
Comment (susnux) [inline lib/Controller/ShareApiController.php:58]:
```suggestion

```

Comment (susnux) [reply lib/Controller/ShareApiController.php:58]: (white space - best run `cs:fix` locally)

File: lib/Service/SubmissionService.php [hunk 2]
```diff
@@ -60,66 +58,24 @@
...
+		private IL10N $l10n,
+		private LoggerInterface $logger,
+		private IUserManager $userManager,
 		IUserSession $userSession,
-		IMailer $mailer,
+		private IMailer $mailer,
 		private ITempManager $tempManager,
 		private FormsService $formsService,
 		private IUrlGenerator $urlGenerator,
 		private UploadedFileMapper $uploadedFileMapper,
```
Comment (susnux) [inline lib/Service/SubmissionService.php:77]: Not a blocker but as you already changed the order here, maybe make it a bit nicer and reorder the `UploadedFileMapper` to the other mappers (so e.g. below the answer mapper)?

**Review by susnux — APPROVED** [review]
Except for the whitespace issue this looks good!

**Review by susnux — APPROVED** [review]

**Pull request merged**